app/Jobs/Copyright/DeleteEntityImage.php

```diff
@@ -0,0 +1,87 @@
+<?php
+
+namespace App\Jobs\Copyright;
+
+use App\Models\Campaign;
+use Illuminate\Bus\Queueable;
+use Illuminate\Contracts\Queue\ShouldQueue;
+use Illuminate\Foundation\Bus\Dispatchable;
+use Illuminate\Queue\InteractsWithQueue;
+use Illuminate\Queue\SerializesModels;
+use Illuminate\Support\Facades\Log;
+use Illuminate\Http\Request;
+use App\Services\ImageService;
+use App\Models\Entity;
+
+class DeleteEntityImage implements ShouldQueue
+{
+    use Dispatchable;
+    use InteractsWithQueue;
+    use Queueable;
+    use SerializesModels;
+
+    protected $entityId;
+    protected $removeHeader;
+    protected $removeImage;
+
+    /**
+     * Create a new job instance.
+     *
+     * @return void
+     */
+    public function __construct(Request $request)
+    {
+        $this->entityId = $request->post('entity');
+        $this->removeHeader = $request->post('remove_header');
+        $this->removeImage = $request->post('remove_image');
+    }
+
+    /**
+     * Execute the job.
+     *
+     * @return void
+     */
+    public function handle()
+    {
+        // Most basic setup, the child has an image, otherwise, might have a gallery image
+        if ($this->removeImage) {
+            $this->deleteImage('image');
+        }
+        if ($this->removeHeader) {
+            $this->deleteImage('header_image');
+        }
+
+        Log::info('Removed image or header from entity #' . $this->entityId . ' for copyrigh reasons');
+    }
+
+    private function deleteImage(string $field)
+    {
+        $entity = Entity::find($this->entityId);
+
+        if (empty($entity) || empty($entity->child)) {
+            // Entity was deleted
+            return;
+        }
+        $service = app()->make(\App\Services\CampaignService::class);
+        $campaign = Campaign::find($entity->campaign_id);
+
+        $service->removedImage($campaign, $entity);
+        $child = $entity->child;
+
+        if (!empty($entity->child->image)) {
```

Comment:
The logic needs to be the other way round: gallery images are a superboosted campaign feature. If the campaign is superboosted, we need to remove the gallery images first, so that if the user unboosts their campaign, it falls back to the "uploaded" image in case they have uploaded one.

With the current way, if the campaign used to be superboosted and had an image, it would potentially delete a gallery image that could be attached to many other entities. To avoid that, we can only delete the gallery images if the campaign is superboosted `if ($campaign->superboosted())`